docs/DeveloperGuide.md

```diff
@@ -1,11 +1,43 @@
 # Developer Guide
 
+### Table of Contents
```

Comment:
This table of contents is super handy but it will need to be updated periodically

Reply:
Thanks for creating this table of contents! It makes it really easy for users to navigate now.